Sources/cggen/main.swift

```diff
@@ -1,51 +1,83 @@
-  parser.newString(cggenSupportHeaderPathKey)
-  parser.newString(moduleKey)
-  parser.newFlag(verboseFlagKey)
-  parser.parse()
-  return Args(
-    objcHeader: parser.string(at: objcHeaderKey),
-    objcPrefix: parser.string(at: objcPrefixKey),
-    objcImpl: parser.string(at: objcImplKey),
-    objcHeaderImportPath: parser.string(at: objcHeaderImportPathKey),
-    objcCallerPath: parser.string(at: objcCallerPathKey),
-    callerScale: parser.double(at: callerScaleKey) ?? 1,
-    callerPngOutputPath: parser.string(at: callerPngOutputPathKey),
-    generationStyle: parser.string(at: generationStyleKey),
-    cggenSupportHeaderPath: parser.string(at: cggenSupportHeaderPathKey),
-    module: parser.string(at: moduleKey),
-    verbose: parser.getFlag(verboseFlagKey),
-    files: parser.getArgs()
-  )
+
+  public mutating func run() throws {
+    Logger.shared.setLevel(level: verbose)
+    var stopwatch = StopWatch()
+    let files = self.files.map(URL.init(fileURLWithPath:))
+    let images = try generateImages(from: files)
+
+    log("Parsed in: \(stopwatch.reset())")
+    let style = generationStyle.flatMap(GenerationParams.Style.init(rawValue:)) ?? .plain
+    let params = GenerationParams(
+      style: style,
+      prefix: objcPrefix,
+      module: module
+    )
+
+    if let objcHeaderPath = objcHeader {
+      let headerGenerator = ObjcHeaderCGGenerator(params: params)
+      let fileStr = headerGenerator.generateFile(images: images)
+      try! fileStr.write(toFile: objcHeaderPath, atomically: true, encoding: .utf8)
+      log("Header generated in: \(stopwatch.reset())")
+    }
+
+    if let objcImplPath = objcImpl {
+      let headerImportPath = objcHeaderImportPath
+      let implGenerator = ObjcCGGenerator(
+        params: params,
+        headerImportPath: headerImportPath
+      )
+      let fileStr = implGenerator.generateFile(images: images)
+      try! fileStr.write(toFile: objcImplPath, atomically: true, encoding: .utf8)
+      log("Impl generated in: \(stopwatch.reset())")
+    }
+
+    if case .swiftFriendly = params.style, let path = cggenSupportHeaderPath {
+      try! params.cggenSupportHeaderBody.renderText()
+        .write(toFile: path, atomically: true, encoding: .utf8)
+      log("cggen_support was generated in: \(stopwatch.reset())")
+    }
+
+    if let objcCallerPath = objcCallerPath,
+       let pngOutputPath = callerPngOutputPath,
+       let headerImportPath = objcHeaderImportPath {
+      let callerGenerator = ObjcCallerGen(
+        headerImportPath: headerImportPath,
+        scale: callerScale.cgfloat,
+        allowAntialiasing: callerAllowAntialiasing,
+        prefix: objcPrefix,
+        outputPath: pngOutputPath
+      )
+      let fileStr = callerGenerator.generateFile(images: images)
+      try! fileStr.write(toFile: objcCallerPath, atomically: true, encoding: .utf8)
+      log("Caller generated in: \(stopwatch.reset())")
+    }
+  }
 }
 
-try runCggen(with: parseArgs())
```

Comment:
I don't really understand why this needed, can you explain please? Code is now duplicated between tests and real code for no apparent reason, and there is no clear entrance function to call cggen and a lot of public keywords scattered all around.